### DISCUSSION TITLE
NextAuth JWT + custom AWS lambdas authorizer

### DISCUSSION THREAD

**Discussion by pedrommarquesm** [nextauthjs/next-auth#5570] [Help]
Hello.

While using the JWT session strategy, I've now stumbled upon a (possible) problem, that so far I couldn't solve.

Assuming that:
1) I am using a custom backend for data management (aws lambdas + rds)
2) this backend is capable of decoding and decrypting NextAuth generated JWT's (having access to its secret)

**- How can I send this JWT to my custom backend via Authorization Header?**

I've been going round and round on how to "expose" the JWT to the client, so I can make an HTTP request on the client side to the custom backend.
Don't really know if this is so difficult to avoid messing with the JWT on the client side or not, but since it is encrypted...

The only way I can think of letting the server know who the user making the request is, is this way (by decoding the JWT and accessing the user id).

Any help would be great,
Thanks!

**Comment by iWanheda** [comment]
I have the exact same question!

**Reply by pedrommarquesm** [reply]
I found out this question with a good solution (not really an answer): https://www.reddit.com/r/nextjs/comments/mta4yk/nextauth_with_custom_backend/

Still can't get my head around on why I can't use the JWT created by Next-Auth, since I am able to decode and decrypt it on my custom backend...

So basically, a possible solution, is to create a JWT (either by requesting a JWT to your custom server, or by creating directly on the SignIn callback) and transfer it in the user object: `user.my-custom-jwt = generateCustomJwt(...)`

If there's no better suggestion, I will have to implement it this way...

**Comment by iWanheda** [comment]
I don't seem to be able to get a consistent idea of how this works, do you have discord by any chance?
